activation/atxdb.go

```diff
@@ -720,11 +721,13 @@ func (db *DB) GetFullAtx(id types.ATXID) (*types.ActivationTx, error) {
 	if err != nil {
 		return nil, err
 	}
-	header, err := db.GetAtxHeader(id)
+	/*header, err := db.GetAtxHeader(id)
 	if err != nil {
 		return nil, err
 	}
 	atx.ActivationTxHeader = header
+
+	*/
```

Comment:
This change is problematic - the resulting ATX will have no ID set. The `GetAtxHeader` method sets the ID field in the header, which is a private field (lowercase) so isn't stored to disk.

Reply:
setting stuff in getter usually leads to these kind of problems... we should think about a way to solve this (I can't think of how right now)

Reply:
the getter is responsible for deserializing the header and returning an initialized Go structure. The ID field isn't stored on disk since it's essentially just a cache. Setting it in the getter avoids the need to recalculate the ID (which is a hash of the ATX) when we already know it (that's the key of the database entry).

I don't think that the way it was implemented is problematic and I don't see the motivation for this change here. Maybe if I understand the motivation I can suggest a solution.